### DISCUSSION TITLE
Filtering content in a table

### DISCUSSION THREAD

**Discussion by samarthpuri** [twilio-labs/paste#1635] [Ideas] [answered]
I'm currently working on a filtering system for contents within a table. So far I created a spreadsheet-like filtering tab system since our specs require searching for content within a column. 
<img width="461" alt="Screen Shot 2021-07-02 at 1 53 21 PM" src="https://user-images.githubusercontent.com/23617133/124312164-de5ba200-db3c-11eb-9411-53369390b21e.png">

I know traditionally we've used a top-level search/filters like this
<img width="611" alt="Screen Shot 2021-07-02 at 2 00 05 PM" src="https://user-images.githubusercontent.com/23617133/124312747-d18b7e00-db3d-11eb-8e92-34f2ee7f3b19.png">

But I was curious if there were existing systems that were utilizing a spreadsheet-like filtering tab or any alternatives to the top-level search/filters?

**Comment by nkrantz** [comment, marked as answer]
Hi @samarthpuri!

You're right that the second screenshot has been the standard table filter pattern for most teams. Before we look for other examples/advice for alternative patterns, it would help to know a bit more about what you're trying to achieve and _why_ the typical pattern doesn't work for your use case.

Thanks!

**Reply by samarthpuri** [reply]
Hi nkrantz! 

So I currently have 4-5 columns that I need to filter or search content with. Based on the traditional top-level search/filters, it's a little tight to add 4-5 search or filter bars given the available spacing.

**Reply by nkrantz** [reply]
Gotcha! We don't currently have a recommended pattern for multiple filters on a table, but @corinne-nagel is currently working on the filter group pattern contribution. I'd recommend reaching out to her to collaborate!
